src/solver/optimisation/adequacy_patch_csr/construct_problem_constraints_RHS.cpp

```diff
@@ -63,7 +64,7 @@ void HourlyCSRProblem::setRHSnodeBalanceValue()
 
     for (int Area = 0; Area < problemeHebdo_->NombreDePays; Area++)
```

Comment:
Yes, but currently we have 2 layers
- One C++ layer defined in libs/antares/study
- One C layer defined in solver/ SIM

The C++ structures are translated into SIM, so that OPT/SIM doesn't use C++ structures. Do we want to change this logic ?

Reply:
When I made this remark, I was worried about the dependency that the **CSR problem** has to the **weekly problem**, especially to the **adq patch** part of the **weekly problem**.
I still am.
After your last answer, I see that the weekly problem also contains some information about areas, links, number of clusters, ... as though this pieces of information already exist in the study. 
So it looks like some code duplication. 
But I don't think now is the time to address this (supposed) code duplication in the weekly problem.
That said, we could avoid the **weekly problem** dependency in the **CSR problem** by directly use **study->areas**

Reply:
OK, agreed. Let's not use SIM in OPT_AdequacyPatch. See #1162